### ISSUE TITLE
Intelligent menu hiding

### ISSUE THREAD

**Issue by dankamongmen** [dankamongmen/notcurses#307]
Dependent on #179 and then #306. We ought support "intelligent menu hiding":

- [ ] the menu is initially visible, but disappears shortly after creation
- [ ] when the mouse is taken to the menu's boundary, it appears again
- [ ] if the mouse goes away, or escape is pressed, or the menu is used, it disappears
- [ ] if we wanna be really cool, fade in/out

low priority.
